File: frontend/src/types/report.ts
export type ReportPeriod = "7d" | "30d" | "90d";
export type ReportView = "daily" | "weekly";

export interface ReportRange {
  period?: ReportPeriod;
  startDate?: string;
  endDate?: string;
}

export interface CalorieTrend {
  points: Array<{
    period: string;
    calories: number;
  }>;
}

export interface MacroTrend {
  points: Array<{
    period: string;
    protein_g: number;
    carbs_g: number;
    fat_g: number;
  }>;
}

export interface MicronutrientSummary {
  nutrients: Array<{
    name: string;
    amount: number;
  }>;
}

export interface GoalMetric {
  target: number;
  actual: number;
  remaining: number;
  percentage: number;
}

export interface GoalComparison {
  date: string;
  end_date: string | null;
  has_active_goal: boolean;
  calories: GoalMetric | null;
  protein_g: GoalMetric | null;
  carbs_g: GoalMetric | null;
  fat_g: GoalMetric | null;
}

File: backend/app/schemas/report.py
from datetime import date

from pydantic import BaseModel


class CalorieTrendPoint(BaseModel):
    period: str
    calories: int


class CalorieTrendResponse(BaseModel):
    points: list[CalorieTrendPoint]


class MacroTrendPoint(BaseModel):
    period: str
    protein_g: float
    carbs_g: float
    fat_g: float


class MacroTrendResponse(BaseModel):
    points: list[MacroTrendPoint]


class MicronutrientTotal(BaseModel):
    name: str
    amount: float


class MicronutrientSummaryResponse(BaseModel):
    nutrients: list[MicronutrientTotal]


class GoalMetric(BaseModel):
    target: float
    actual: float
    remaining: float
    percentage: float


class GoalComparisonResponse(BaseModel):
    date: date
    end_date: date | None = None
    has_active_goal: bool
    calories: GoalMetric | None
    protein_g: GoalMetric | None
    carbs_g: GoalMetric | None
    fat_g: GoalMetric | None
